src/credentials/CHIPCert.h

```diff
@@ -641,6 +641,81 @@ CHIP_ERROR ConvertX509CertToChipCert(const uint8_t * x509Cert, uint32_t x509Cert
+ * @param x509CertLen     The length of the converted certificate.
+ *
+ * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+ **/
+CHIP_ERROR GenerateSignedX509CertFromChipCert(const uint8_t * chipCert, uint32_t chipCertLen, Crypto::P256Keypair & keypair,
+                                              uint8_t * x509CertBuf, uint32_t x509CertBufSize, uint32_t & x509CertLen);
+
+struct X509CertRequestParams
+{
+    int64_t mSerialNumber;
+    uint64_t mIssuer;
+    uint32_t mValidityStart;
+    uint32_t mValidityEnd;
+    bool mHasFabricID;
+    uint64_t mFabricID;
+    bool mHasNodeID;
+    uint64_t mNodeID;
+};
+
+/**
+ * @brief Generate a new X.509 DER encoded Root CA certificate
+ *
+ * @param requestParams   Certificate request parameters.
+ * @param issuerKeypair   The certificate signing key
+ * @param x509CertBuf     Buffer to store signed certificate in X.509 DER format.
+ * @param x509CertBufSize The size of the buffer to store converted certificate.
+ * @param x509CertLen     The length of the converted certificate.
+ *
+ * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+ **/
+CHIP_ERROR NewRootX509Cert(X509CertRequestParams & requestParams, Crypto::P256Keypair & issuerKeypair, uint8_t * x509CertBuf,
+                           uint32_t x509CertBufSize, uint32_t & x509CertLen);
+
+/**
+ * @brief Generate a new X.509 DER encoded Intermediate CA certificate
+ *
+ * @param requestParams   Certificate request parameters.
+ * @param subject         The requested subject ID
+ * @param subjectPubkey   The public key of subject
+ * @param issuerKeypair   The certificate signing key
+ * @param x509CertBuf     Buffer to store signed certificate in X.509 DER format.
+ * @param x509CertBufSize The size of the buffer to store converted certificate.
+ * @param x509CertLen     The length of the converted certificate.
+ *
+ * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+ **/
+CHIP_ERROR NewICAX509Cert(X509CertRequestParams & requestParams, uint64_t subject, const Crypto::P256PublicKey & subjectPubkey,
+                          Crypto::P256Keypair & issuerKeypair, uint8_t * x509CertBuf, uint32_t x509CertBufSize,
+                          uint32_t & x509CertLen);
+
+/**
+ * @brief Generate a new X.509 DER encoded Node operational certificate
+ *
+ * @param requestParams   Certificate request parameters.
+ * @param rootCASigner    Indicates if the signer is a root CA or an intermediate CA
+ * @param subjectPubkey   The public key of subject
+ * @param issuerKeypair   The certificate signing key
+ * @param x509CertBuf     Buffer to store signed certificate in X.509 DER format.
+ * @param x509CertBufSize The size of the buffer to store converted certificate.
+ * @param x509CertLen     The length of the converted certificate.
+ *
+ * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+ **/
+CHIP_ERROR NewNodeOperationalX509Cert(X509CertRequestParams & requestParams, bool rootCASigner,
```

Comment:
use enums over bools

`NewNodeOperationalX509Cert(params, true, ...)`

is less readable than

`NewNodeOperationalX509Cert(parans, kSignerIsRootCa, ...)`